src/opensearch_sdk_py/api/extension.py

```diff
@@ -9,14 +9,8 @@
 
 # https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/Extension.java
 
-from abc import ABC, abstractmethod
+from abc import ABC
 
 
 class Extension(ABC):
-    @property
-    @abstractmethod
-    def implemented_interfaces(self) -> list[tuple]:
-        """
-        Implementer should return a list of tuples containing the implemented interface (such as
-        Extension, ActionExtension, etc.) and the implementing class.
-        """
+    pass
```

Comment:
I see two `extension.py` classes, this one and one in the parent directory.   The parent one inherits from this one which seems somehow intentional, but ActionExtension (in this directory) also inherits from it.  Clearly there's some magic in how we're doing implemented extensions, but package structure isn't making it very clear.  Some comments on what's going on may be helpful.

Reply:
I was trying to have a base interface for classes like `ActionExtension`, but this is actually unnecessary. Removed.